neuralpredictors/measures/np_functions.py

```diff
@@ -12,10 +14,157 @@ def corr(y1, y2, axis=-1, eps=1e-8, **kwargs):
         y2:      second NumPy array
         axis:    dimension(s) along which the correlation is computed. Any valid NumPy axis spec works here
         eps:     offset to the standard deviation to avoid exploding the correlation due to small division (default 1e-8)
-        **kwargs: passed to final numpy.mean operatoin over standardized y1 * y2
+        **kwargs: passed to final numpy.mean operation over standardized y1 * y2
 
     Returns: correlation array
     """
+
     y1 = (y1 - y1.mean(axis=axis, keepdims=True)) / (y1.std(axis=axis, keepdims=True, ddof=0) + eps)
     y2 = (y2 - y2.mean(axis=axis, keepdims=True)) / (y2.std(axis=axis, keepdims=True, ddof=0) + eps)
     return (y1 * y2).mean(axis=axis, **kwargs)
+
+
+def oracle_corr_corrected(repeated_outputs):
+    """
+    Compute the corrected oracle correlations per neuron.
+
+    Args:
+        repeated_outputs (array-like): numpy array with shape (images, repeats, responses), or a list of lists of
+            responses (with len: num of repeats) per image.
+
+    Returns:
+        array: Corrected oracle correlations per neuron
+    """
+
+    if len(repeated_outputs.shape) == 3:
+        var_noise = repeated_outputs.var(axis=1).mean(0)
+        var_mean = repeated_outputs.mean(axis=1).var(0)
+    else:
+        var_noise, var_mean = [], []
+        for repeat in repeated_outputs:
+            var_noise.append(repeat.var(axis=0))
+            var_mean.append(repeat.mean(axis=0))
+        var_noise = np.mean(np.array(var_noise), axis=0)
+        var_mean = np.var(np.array(var_mean), axis=0)
+    return var_mean / np.sqrt(var_mean * (var_mean + var_noise))
+
+
+def oracle_corr(repeated_outputs):
+    """
+    Compute the oracle correlations per neuron.
+
+    Args:
+        repeated_outputs (array-like): numpy array with shape (images, repeats, responses), or a list of lists of
+            responses (with len: num of repeats) per image.
+
+    Returns:
+        array: Oracle correlations per neuron
+    """
+
+    if len(repeated_outputs.shape) == 3:
```

Comment:
what is this checking?

Reply:
@KonstantinWilleke I thought this checks if it is a numpy array or a list, but not sure?

Reply:
repeated_outputs can be an array of arrays (shape = (n_images)), or a single array of shape `(images, repeats, responses)`. That's what the check is for. Since the number of repeats is usually variable, the first type is more common.

Reply:
why not just use the code under `else`? that would work for both cases, right?

Reply:
Hm ok - then this check is rather crude for that but I'm ok with it. Could you add an explicit comment to state what this is checking?

Reply:
why not just use the code under `else`? that would work for both cases, right?

Reply:
Thank you guys, I'll remove the code under if and just keep the code under else